lib/didComponents.js

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
+/**
+ * Using a colon (:) as the delimiter, split the identifier
+ * into its components: a scheme, a method, a version, and a
+ * multibaseValue. If there are only three components set the
+ * version to the string value 1 and use the last value as
+ * the multibaseValue.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.did - A did as a string.
+ *
+ * @returns {object} The components of the did.
+*/
+export const getDidKeyComponents = ({did}) => {
```

Comment:
Abstraction here feels wrong... getting method-specific things should probably be placed in method-specific libraries.

Reply:
Totally agree here, this should probably be in `did-method-key-js`. I made an issue to start discussion on this: https://github.com/digitalbazaar/did-method-key/issues/46

There are a lot of validation stuff included here that should be in `did-method-key-js` `ed25519-verification-key-2020` and possibly `x25519`.

Reply:
Marking this as resolved as `getDidKeyComponents` is no longer in this library and the way we parse did keys has changed. See https://github.com/digitalbazaar/did-method-key/pull/47 for how did keys are parsed now.